salt/modules/gpg.py

```diff
@@ -852,7 +853,7 @@ def trust_key(keyid=None,
 
     cmd = 'echo "{0}:{1}" | {2} --import-ownertrust'.format(_cmd_quote(fingerprint),
                                                             _cmd_quote(NUM_TRUST_DICT[trust_level]),
-                                                            _cmd_quote(_check_gpg()))
+                                                            _check_gpg())
```

Comment:
We should not be piping anything into a command, instead we should be using the ``stdin`` argument to the ``cmd.run`` functions. Also, we should not be using ``_cmd_quote()``, it's prone to errors. We should instead be formulating a command as a list of arguments and using ``python_shell=False``.